clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetsRequestManager.java

```diff
@@ -283,14 +283,15 @@ private CompletableFuture<Void> updatePositionsWithOffsets(long deadlineMs) {
         cacheExceptionIfEventExpired(result, deadlineMs);
 
         CompletableFuture<Void> updatePositions;
+        final Set<TopicPartition> initializingPartitions = subscriptionState.initializingPartitions();
         if (commitRequestManager != null) {
-            CompletableFuture<Void> refreshWithCommittedOffsets = initWithCommittedOffsetsIfNeeded(deadlineMs);
+            CompletableFuture<Void> refreshWithCommittedOffsets = initWithCommittedOffsetsIfNeeded(initializingPartitions, deadlineMs);
```

Comment:
Perhaps `initWithCommittedOffsetsIfNeeded` could return a `CompletableFuture<Set<TopicPartition>>` to include the initializing partitions, as the `refreshOffsets` method can provide this information.

Reply:
> Am I maybe missing an issue with this approach of the explicit input param or is it that you find the return value clearer?

I was considering that initWithCommittedOffsetsIfNeeded might adjust the initializing partitions, similar to KAFKA-16966, but that isn't happening at the moment. I'm +1 on the current approach, as it’s simpler and more readable.